### ISSUE TITLE
Dynamic_batching not working correctly with tensorflow models

### ISSUE THREAD

**Issue by farzanehnakhaee70** [triton-inference-server/server#3869]
**Description**
I'm deploying fast-speech model on triton. However, for triton version>=21.10, there is a problem with dynamic batching (this problem is not existed for older versions). Also whenever I disable the dynamic batching, it also works perfectly in all triton versions.

Is this one related to this issue? #3624

**Triton Information**
The problem existed in Triton 21.11 and 21.12.
Also I'm using triton container

**To Reproduce**

This error occurs for fast-speech2 model which is in TensorFlow framework. Here is the link to the model.
https://github.com/TensorSpeech/TensorFlowTTS/tree/master/examples/fastspeech2

**Expected behavior**
Generally when the dynamic batching is activated, randomly in sometimes, it generated outputs with unacceptable shape which results in error for the following procedures.

**Comment by CoderHam** [comment]
@farzanehnakhaee70 can you share a minimal repro for the same?
@tanmayv25 had we recently changed/fixed the dynamic batcher?

**Comment by tanmayv25** [comment]
> However, for triton version>=21.10, there is a problem with dynamic batching (this problem is not existed for older versions). 

Can you describe what is this problem? Are you seeing the batch size not large enough as what is described in the issue #3624 . If yes, then it will be fixed in our 22.01 release.  You can try [this ](https://github.com/triton-inference-server/server/pull/3789) fix and see if it solves your problem.

**Comment by tanmayv25** [comment]
> Generally when the dynamic batching is activated, randomly in sometimes, it generated outputs with unacceptable shape which results in error for the following procedures.

Ahh.. I see. This should not be happening. Please try the 22.01 release when it is out or try to build triton with the fix described and see if it solves your problem. Can you share your config.pbtxt for the model?

**Comment by farzanehnakhaee70** [comment]
Hi @tanmayv25 
Thanks a lot for your response. Yeah, the problem is solved in version 22.01.
Here is my config file as well:
```
name: "fastspeech-2-ace"
platform: "tensorflow_savedmodel"
max_batch_size: 16

input [
  {
    name: "energy_ratios"
    data_type: TYPE_FP32
    dims: [ 1 ]
    reshape: { shape: [ ] }
  },
  {
    name: "f0_ratios"
    data_type: TYPE_FP32
    dims: [ 1 ]
    reshape: { shape: [ ] }
  },
  {
    name: "input_ids"
    data_type: TYPE_INT32
    dims: [ -1 ]
  },
  {
    name: "speaker_ids"
    data_type: TYPE_INT32
    dims: [ 1 ]
    reshape: { shape: [ ] }
  },
  {
    name: "speed_ratios"
    data_type: TYPE_FP32
    dims: [ 1 ]
    reshape: { shape: [ ] }
  }

]
output [
  {
    name: "output_0"
    data_type: TYPE_FP32
    dims: [ -1, 80 ]
  },
  {
    name: "output_1"
    data_type: TYPE_FP32
    dims: [ -1, 80 ]
  },
  {
    name: "output_2"
    data_type: TYPE_INT32
    dims: [ -1 ]
  },
  {
    name: "output_3"
    data_type: TYPE_FP32
    dims: [ -1 ]
  },
  {
    name: "output_4"
    data_type: TYPE_FP32
    dims: [ -1 ]
  }
]

dynamic_batching {
    max_queue_delay_microseconds: 500
  }
  
instance_group [
    {
      count: 1
      kind: KIND_GPU
      gpus: [ 0 ]
    }
]

model_warmup [
{
  name: "warmup"
  batch_size: 2
  inputs[
    {
      key: "energy_ratios"
      value: {
        data_type: TYPE_FP32
        dims: [ 1 ]
        zero_data : True
      }
    },
    {
      key: "f0_ratios"
      value: {
        data_type: TYPE_FP32
        dims: [ 1 ]
        zero_data : True
      }
    },
    {
      key: "input_ids"
      value: {
        data_type: TYPE_INT32
        dims: [ 3 ]
        zero_data : True
      }
    },
    {
      key: "speaker_ids"
      value: {
        data_type: TYPE_INT32
        dims: [ 1 ]
        zero_data : True
      }
    },
    {
      key: "speed_ratios"
      value: {
        data_type: TYPE_FP32
        dims: [ 1 ]
        zero_data : True
      }
    }
  ]
}
]

```

**Comment by CoderHam** [comment]
@farzanehnakhaee70 thank you for confirming the same. Closing the issue.
